src/app/db/WDCReader.ts

```diff
@@ -188,8 +188,20 @@ export default class WDCReader {
 		if (rawDbd !== null)
 			structure = new DBDParser(rawDbd).getStructure(buildID, layoutHash);
 
-		// No cached definition, download updated DBD and check again.
+		let forceDownload = false;
 		if (structure === null) {
+			// No definition, force redownload
+			forceDownload = true;
+		} else {
+			// If definition has placeholder fields, force redownload
+			for (const field of structure.fields) {
+				if (field.name.startsWith('Field_'))
+					forceDownload = true;
+			}
+		}
```

Comment:
This could potentially be reduced to something like

```js
const forceDownload = structure?.fields.some(e => e.name.startsWith('Field_')) || false
```

Or if `forceDownload` is only used the once it could be inlined into the `if` statement below.

Reply:
I don't know if that's the exact same behavior as what I wrote, it might very well be but I don't really know what's going on in that line, sorry.

Reply:
`array.some` returns `true` if at least one item in the array matches the function provided to it. `array.some(e => e.name.startsWith('Field_'))` returns `true` if one item in the array has a name starting with `Field_`.

Since `structure` can be `null`, the `?.` is the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) which short-circuits the statement and returns `null` if `structure` is null or undefined, preventing the following function call from erroring due to a null reference.

Finally, since the optional chain could short-circuit and return null, the `|| false` ensures that the statement is a boolean. Since `null` and `undefined` are falsy, we don't technically need that last bit, but it makes more sense to have `true` and `false` rather than `true`, `false` or `null`.